### DISCUSSION TITLE
[Middleware] How to change global connection in Lucid ORM (AdonisJS 6)?

### DISCUSSION THREAD

**Discussion by brunojoao** [adonisjs/core#4648] [General]
In AdonisJS 6, I'm trying to dynamically change the database connection for a Lucid ORM model based on incoming requests. In previous versions, I could easily switch the connection directly on the model instance using useConnection(params.db_sm).all(). However, this method is no longer available in AdonisJS 6.

I've tried using Database.connection(), Database.use() or Database.call(), but they don't seem to work for switching connections globally and dynamically like before. Could someone please clarify how to achieve dynamic and global connection switching in AdonisJS 6? For example, how can I change the connection based on a request received in a middleware?

Thank you for your help!

**Comment by brunojoao** [comment]
I believe that the entire lucid Database changed, but there is no documentation, there is only db.connection().query, in the ORM there is nothing about it and there is no longer any exchange method.

**Comment by brunojoao** [comment]
![image](https://github.com/adonisjs/core/assets/16447379/12cb948d-3149-49d5-af4d-f5240f0e7b48)

This is the only information about, but nothing about, the Orm model, and overall

**Comment by brunojoao** [comment]
![image](https://github.com/adonisjs/core/assets/16447379/2f767a28-622f-4c8e-a2ee-8b935518e53c)

`        BaseModel.connection = dbTarget`

Solution

Instead of using Database.connection as in v5, use BaseModel.connection, the documentation says not to use it, but for the dynamic case it would be correct, right? In my case, I change databases according to the url parameter, but I couldn't do the same with "db", I did one for each one.

**Reply by Banashek** [reply]
This does seem like the most ergonomic way to do this for now.

My interpretation of the lucid code is that these types of dynamic connection changes should use the `Adapter` interface that can be set on the model dynamically.

There is a `Database.modelAdapter()` method that returns an Adapter that can be passed to the `BaseModel.useAdapter()` function, but I don't see a clean way to change the connection of the `Database` object for the purpose of getting an `Adapter` with a different connection. 

There's also a `useConnection` method on the `BaseModelImpl` but that's not surfaced up to models.

The `Adapter` class is also not exposed outside the package, so you can't new up one of those.

This seems like a potential area for improvement unless I've missed some obvious way of swapping this at runtime.
